pkg/handlers/search_files.go

```diff
@@ -0,0 +1,135 @@
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/artmoskvin/hide/pkg/model"
+	"github.com/artmoskvin/hide/pkg/project"
+)
+
+// TODO: add string enums for search types
+
+const (
+	queryKey      = "query"
+	searchTypeKey = "search"
+)
+
+type SearchFilesHandler struct {
+	ProjectManager project.Manager
+}
+
+func (h SearchFilesHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	projectID, err := getProjectID(r)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Invalid project ID: %s", err), http.StatusBadRequest)
+		return
+	}
+
+	query := r.URL.Query().Get(queryKey)
+	if query == "" {
+		http.Error(w, "Query not specified", http.StatusBadRequest)
+		return
+	}
+
+	typ := r.URL.Query().Get(searchTypeKey)
+
+	check, err := getChecker(typ, query)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Bad query: %s", err), http.StatusInternalServerError)
+	}
+
+	listFiles := func(ctx context.Context, showHidden bool) ([]*model.File, error) {
+		return h.ProjectManager.ListFiles(ctx, projectID, showHidden)
+	}
+
+	result, err := findInFiles(r.Context(), listFiles, check)
+	if err != nil {
+		var projectNotFoundError *project.ProjectNotFoundError
+		if errors.As(err, &projectNotFoundError) {
+			http.Error(w, projectNotFoundError.Error(), http.StatusNotFound)
+			return
+		}
+
+		http.Error(w, fmt.Sprintf("Failed to search: %s", err), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(result)
+}
+
+func getChecker(typ, query string) (check func(s string) bool, err error) {
+	switch typ {
+	case "exact":
+		return exactSearch(query)
+	case "grep":
+		return grepSearch(query)
+	default:
+		return caseInsensitiveSearch(query)
+	}
+}
```

Comment:
I'd say these are more like search options, not search type. The search type is the same – content search.

Reply:
In this case non, just symmetric to the signature of function called here.